### PR TITLE
ENH: multi-gpu Random Forest [SPMD]

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description

This PR is enhancement for #949, #1175:
* Adding `spmd` and `spmd.ensemble` module
* multi-GPU SPMD interface for Random Forest Onedal4py implementation
* Examples for RF spmd

### CODE REVIEW

**Comment by samir-nasibli** [comment]
Waiting #1059 #1064 and https://github.com/oneapi-src/oneDAL/pull/2156

File: onedal/common/mpi_primitives.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+ONEDAL_PY_INIT_MODULE(mpi_primitives) {
+    //using mpi_initializer_t = mpi_initializer;
+    py::class_<mpi_initializer>(m, "mpi_initializer")
+        .def(py::init())
+        .def("init", &mpi_initializer::init)
+        .def("fini", &mpi_initializer::fini);
+}
+} // namespace oneapi::dal::python
+#endif
```
Comment (napetrov) [inline onedal/common/mpi_primitives.cpp:43]:
```suggestion
#endif

```

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Rebase on last changes in sklearnex.
This PR should pass github CI.

**Comment by samir-nasibli** [comment]
Outdated. Implemented on #1214

**Pull request closed without merging**